sklearn/covariance/elliptic_envelope.py

```diff
@@ -97,79 +95,92 @@ def __init__(self, store_precision=True, assume_centered=False,
 
     def fit(self, X, y=None):
-        """Fit the EllipticEnvelope model with X.
+        """Fit the EllipticEnvelope model.
 
         Parameters
         ----------
-        X : numpy array or sparse matrix of shape [n_samples, n_features]
+        X : numpy array or sparse matrix, shape (n_samples, n_features).
             Training data
         y : (ignored)
         """
         super(EllipticEnvelope, self).fit(X)
-        self.threshold_ = sp.stats.scoreatpercentile(
-            self.dist_, 100. * (1. - self.contamination))
+        self.offset_ = sp.stats.scoreatpercentile(
+            -self.dist_, 100. * self.contamination)
         return self
 
-    def decision_function(self, X, raw_values=False):
+    def decision_function(self, X, raw_values=None):
         """Compute the decision function of the given observations.
 
         Parameters
         ----------
         X : array-like, shape (n_samples, n_features)
 
-        raw_values : bool
+        raw_values : bool, optional
             Whether or not to consider raw Mahalanobis distances as the
             decision function. Must be False (default) for compatibility
             with the others outlier detection tools.
 
+        .. deprecated:: 0.20
+            ``raw_values`` has been deprecated in 0.20 and will be removed
+            in 0.22.
+
         Returns
         -------
+
         decision : array-like, shape (n_samples, )
             Decision function of the samples.
-            It is equal to the Mahalanobis distances if `raw_values`
-            is True. By default (``raw_values=False``), it is equal
-            to the cubic root of the shifted Mahalanobis distances.
-            In that case, the threshold for being an outlier is 0, which
-            ensures a compatibility with other outlier detection tools
-            such as the One-Class SVM.
+            It is equal to the shifted Mahalanobis distances.
+            The threshold for being an outlier is 0, which ensures a
+            compatibility with other outlier detection algorithms.
 
         """
-        check_is_fitted(self, 'threshold_')
-        X = check_array(X)
-        mahal_dist = self.mahalanobis(X)
-        if raw_values:
-            decision = mahal_dist
-        else:
-            transformed_mahal_dist = mahal_dist ** 0.33
-            decision = self.threshold_ ** 0.33 - transformed_mahal_dist
+        check_is_fitted(self, 'offset_')
+        negative_mahal_dist = self.score_samples(X)
 
-        return decision
+        # raw_values deprecation:
+        if raw_values is not None:
+            warnings.warn("raw_values parameter is deprecated in 0.20 and will"
+                          " be removed in 0.22.", DeprecationWarning)
 
-    def predict(self, X):
-        """Outlyingness of observations in X according to the fitted model.
+            if not raw_values:
+                return (-self.offset_) ** 0.33 - (-negative_mahal_dist) ** 0.33
```

Comment:
Minor: this line is not covered by tests.

Reply:
yes but this line of code is only used if the user set the deprecated `raw_values` parameter... (this line will be removed in 0.22)